### DISCUSSION TITLE
Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren

### DISCUSSION THREAD

**Discussion by RolfGS** [devemux86/cruiser#450] [Q&A] [answered]
Ich habe eine neue Route in Kurviger am PC geplant, im ".kurviger" Format in meinen Download Ordner exportiert und mit Drag & Drop in meinen Google Drive Cloudspeicher gezogen. Wenn ich jetzt Cruiser im Samsung S10e öffne und-->Routing-->Importiere Route anwähle (oder -->meien Orte-->Routen-->Importieren) , dann kann ich nicht mehr auf den Google Cloud Drive zugreifen. 
Es wird mir nur "Öffnen aus "Zuletzt verwendet o. Bilder o. Videos o. Audio o. Dokumente o. Downloads" angeboten.
Was macche ich bitte falsch ?
Gruss,
Rolf Reitenbach

**Comment by RolfGS** [comment, marked as answer]
Hello, found the solution: needed reactivate Google Drive in Playstore. No action more, thanks !

**Comment by devemux86** [comment]
@RolfGS 

Cruiser opens the Android file browser to import files. It has settings in the top right menu.
The left drawer shows all storage locations: internal storage, SD card, cloud, etc.

Cruiser öffnet den Android-Dateibrowser, um Dateien zu importieren. Die Einstellungen befinden sich im Menü oben rechts.
In der linken Schublade werden alle Speicherorte angezeigt: interner Speicher, SD-Karte, Cloud, usw.

<img src="https://github.com/devemux86/cruiser/assets/3484020/5332e2cd-99da-4309-94e7-a135d0aa606f" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/05c8710f-0dbe-4bfa-a8f2-8f128182099c" width="50%"/>

**Reply by bios4** [reply]
> Cruiser opens the Android file browser to import files.

Is there a possibility to allow the user to select a different file browser than the Android one @devemux86?

The file explorer app I'm using is capable to access all of the cloud storages I'm using, plus FTP shares.

My Android browser strangely only allows to access the local file system (either internal or external/SD card). So I have to copy my routes/tracks/... to the local file system first to be able to open them within Cruiser.

If Cruiser would offer the option to use this different file explorer, this extra step would not be required anymore...

**Reply by devemux86** [reply]
@bios4

Android storage now has specific permissions and Android file browser is required to grant file access to the apps.
So it's a bit complicated. :slightly_smiling_face: 

However, as seen in the above images, Google Drive appears in Android browser.
Is it about other cloud services, like Dropbox, etc?

---

Another alternative is to use any file manager (e.g. Total Commander),
press or long press a gpx / json / kurviger file and select open with Cruiser.

**Reply by bios4** [reply]
> Google Drive appears in Android browser.

Unfortunately not on my phone. There only the internal storage and the Cloud storage of my mail provider is offered:

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/5c3559cc-6bdf-4446-8756-723a3dd4fa4c" width="33%">

Altough I have Dropbox, Google Drive, OneDrive and some FTP shares defined on the device in additon, I don't get the possibility to access them in the Android file browser.

**Reply by devemux86** [reply]
@bios4 this is strange as Google Drive is by Google!

What options do you have in the top-right (3-dot) menu in the Android file browser?

Does the Google Files app display the Google Drive?

**Reply by 0709wiwiwi** [reply]
I use  the Cx File Explorer.  
Introduction > Read the manual.
https://cxfileexplorer.com/how-to-use-cx-file-explorer/

You than select Network > set Dropbox or Google Drive or Onedrive.
Browse and select the file that you need and tap open with Cruiser (always).
First time to open a .kurviger file in the cx explorer you may need to select the menu More. 
Than open with or Open as > Other > Than select open with Cruiser (always)

**Reply by 0709wiwiwi** [reply]
To open the Android default file browser directly you can use next.
This app delivers a direct shortcut to the Android default browser.
https://play.google.com/store/apps/details?id=com.marc.files

I have no idea how Samsung handles the original Android ... ;-)

**Reply by bios4** [reply]
> Does the Google Files app display the Google Drive?

It seems to be related to the fact that I don't use the Google Drive app (because my file manager takes over the functionality).

After opening Google Drive once (thanks to @RolfGS for that hint below!), Google Drive is again offered in the Android file browser. But unfortunately not the places I would really need (Dropbox and FTP)...

> I use the Cx File Explorer.

> Browse and select the file that you need and tap open with Cruiser (always).

> Than open with or Open as > Other > Than select open with Cruiser (always)

This is all nice (I use [File Manager Plus (Pro)](https://play.google.com/store/apps/details?id=com.alphainventor.filemanager), which does more or less the same as Cx File Explorer but including access to the otherwise locked Android folders).

Yet I do NOT want to go the path "file manager > Cruiser", but the very OTHER WAY "Cruiser > open files from (a different) file manager" (in fact even from a remote location, Dropbox/FTP). I don't want to set Cruiser as the default app for related files (*.gpx and *.kurviger), because I also handle them with different other apps.

**Reply by bios4** [reply]
BY THE WAY: on my other phone (the one I use on the bike) running Android 10 I can decide which file manager I want to use to select/open my files!!! See below:

<img alt="" src="https://github.com/devemux86/cruiser/assets/145460147/4bfff325-8eef-45d9-a7e9-74de3db88a7b" width="33%">

**Reply by 0709wiwiwi** [reply]
@bios4 
>  I also handle them with different other apps.

(By 2 extra clicks Cx Explorer still offers import in other apps).

> OTHER WAY "Cruiser > open files from (a different) file manager"

 No problem on A10 but not offered anymore on A13 (Cruiser & Locus).
Anyway Locus A13 import still offers "open in Dropbox" or file remotely. 
![image](https://github.com/devemux86/cruiser/assets/54167201/db6a9398-277e-4272-8b4a-3d051205503b)

**Dealing with the A13 issues.**
- Cruiser offers storage in Android/media that is still easily accessible.
- The very latest Locus 4 **Beta** finally also offers this as such to test. 
To tune Locus an access to the app storage is **absolutely** desirable.

**Comment by RolfGS** [comment]
...when I reactivated Google Drive in Playstore it worked again. Now Google Drive shows up in " open with" in Cruiser App and I have access to all files from my Samsung S10e vintage smartphone.

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Comment by RolfGS** [comment]
Hello,
this is my „filemanager“ in Cruiser. On bottom you see that Google Drive Icon is offered. This was not the case before I restarted Goggle Drive thru Play Store. Somehow Goggle Drive must have been
deactivated during the past few 3 weeks …maybe some poer saving feature in my smartphone.. I do not know… . But now its working again and I know how to reactivate.

***@***.***

Von: Mike ***@***.***>
Gesendet: Montag, 19. Februar 2024 19:57
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


BY THE WAY: on my other phone (the one I use on the bike) running Android 10 I can decide which file manager I want to use to select/open my files!!! See below:
Screenshot_20240219-195136_Android.System.jpg (view on web)<https://github.com/devemux86/cruiser/assets/145460147/4bfff325-8eef-45d9-a7e9-74de3db88a7b>

**Comment by RolfGS** [comment]
Hello,

sometimes I do not want to use Cruiser map with navigation but just run the map with my bike position to see where I am going and how curvy the road is ahead myself.

Now I have the problem that my bike (blue dor on map) does not stay in center of the map, its like the dot is moving outside of the screen and I cannot zoom in near the dot when its to far on the corner of the screen.

If I push the center dot on bottom right, the bike symbol is again in middle of the screen, BUT while driving I cannot do this with bike gloves which mean I cannot zoom in adaquately to see more details oft he road ahead of me.

Is there a possibility to automatize the centering of the blue dot if no navigation is running ? Or do you already have planned to integrate such feature – by when would that be available.



Below screenshot shows blue dot relatively in middle, in that case I could adequately zoom in. But if the blue dot is near some side or corner then zooming in is not possible.

***@***.***

**Reply by bios4** [reply]
This does not belong in this topic, does it?

BUT: I assume your're talking about the "follow mode" - so just LONG press the "center dot on bottom right" @RolfGS and you should be good to go...

**Reply by devemux86** [reply]
@RolfGS do you mean the follow location mode? It can be activated similarly to the navigation.

LONG PRESS (not just tap) the location button at bottom right and the follow location mode is enabled.
(then location button is colored)

**Comment by RolfGS** [comment]
Oh, excuse me I just used the email in response mode… .sorry!

Ah, cool, I will try it out. Oh, is it normal that the map goes then in 2D (3D?) mode?


Von: Mike ***@***.***>
Gesendet: Mittwoch, 21. Februar 2024 15:53
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


This does not belong in this topic, does it?

BUT: I assume your're talking about the "follow mode" - so just LONG press the "center dot on bottom right" @RolfGS<https://github.com/RolfGS> and you should be good to go...

**Reply by devemux86** [reply]
> Oh, is it normal that the map goes then in 2D (3D?) mode?

@RolfGS After enabling the follow location, long press the compass to select the orientation mode:
GPS 2D, GPS 3D, etc.

**Comment by RolfGS** [comment]
Thanks, will try it out – did so far not get the idea to press long the button…😊

Von: Emux ***@***.***>
Gesendet: Mittwoch, 21. Februar 2024 15:55
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


@RolfGS<https://github.com/RolfGS> do you mean the follow location mode? It can be activated similarly to the navigation.

LONG PRESS (not just tap) the location button at bottom right and the follow location mode is enabled.
(then location button is colored)

**Comment by RolfGS** [comment]
Perfect ! Many thanks!

Von: Emux ***@***.***>
Gesendet: Mittwoch, 21. Februar 2024 16:07
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


Oh, is it normal that the map goes then in 2D (3D?) mode?

@RolfGS<https://github.com/RolfGS> After enabling the follow location, long press the compass to select the orientation mode:
GPS 2D, GPS 3D, etc.

**Comment by RolfGS** [comment]
Hello,
I have the following problem

  1.  When I want to select a route from below (example), the I get very often the menue under 2. (see below) which does not allow me to load the desired route.
After 10 erratic tries I get the correct menue (see below 3.) , which allows me to load the route.
What am I doing wrong with tapping on the screen?
***@***.***


  1.  Then I get very often the following menue
***@***.***


  1.  BUT NOT this menue
***@***.***


Von: Emux ***@***.***>
Gesendet: Mittwoch, 21. Februar 2024 16:07
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


Oh, is it normal that the map goes then in 2D (3D?) mode?

@RolfGS<https://github.com/RolfGS> After enabling the follow location, long press the compass to select the orientation mode:
GPS 2D, GPS 3D, etc.

**Reply by devemux86** [reply]
@RolfGS 

There are no screenshots in your post, so it is difficult to understand what you mean.

Can you write directly in here and post some screenshots?

**Comment by RolfGS** [comment]
I too often get only screenshot 2 but I need 3 to load the route...hope you can now see the screenshots?

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Reply by devemux86** [reply]
@RolfGS sorry, there are no screenshots, probably because you write via email?

Better to use directly the GitHub website and write your comment and also attach the screenshots.

**Comment by RolfGS** [comment]
...will do, bare with me ...

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Comment by RolfGS** [comment]
Hallo,
hier die 3 Screenshots via GitHub, hoffentlich sind sie nun zu erkennen - can you see the screenshots now ?

<img src="https://github.com/devemux86/cruiser/assets/156905078/8c1abff0-5fe2-4173-9737-0a19c46bca97" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/156905078/12a41915-c0d2-44b7-a211-425500c34c65" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/156905078/687904ea-aedb-4ca0-b7bc-5e66b0cc7a64" width="33%"/>

**Comment by RolfGS** [comment]
Hello, can you see the screenshots now on Github?

Pic 1 shows the routes one of which I need to load
Pic2 shows what menue I mostly get but which does not offer th eoption to load the route
Pic 3 is what I need as it offers the option to load the route. Often I need to tap the screen 10 times to get  Pic3 and not Pic2.

Von: Emux ***@***.***>
Gesendet: Mittwoch, 12. Juni 2024 18:01
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


@RolfGS<https://github.com/RolfGS> sorry, there are no screenshots, probably because you write via email?

Better to use directly the GitHub website and write your comment and also attach the screenshots.

**Reply by devemux86** [reply]
@RolfGS thanks, now we can see the screenshots.

- The menu in image 2 appears when we select **multiple routes** (with tap) to do  actions on all of them.
The selected routes then have a checkbox on their icon at the left.
This menu shows the number of the selected routes "Routen (1)".

- To display the menu in image 3 for a **single route**, _press the menu button with the 3 dots at the right_.
  - Or you can swipe the route to the **right** to load it.

Please let me know if you need more help.

**Comment by RolfGS** [comment]
Ah, thanks, the swiping ist he solution! I did not realize is by myself and it was hard fo my thick fingertip to hit the 3 dots on right correctly.
Thanks again for your quick and competent help – I am „private tourguide“ and today ist the 4th of 5 tour days in Tuscany…

Von: Emux ***@***.***>
Gesendet: Donnerstag, 13. Juni 2024 08:12
An: devemux86/cruiser ***@***.***>
Cc: RolfGS ***@***.***>; Mention ***@***.***>
Betreff: Re: [devemux86/cruiser] Kann Route in .kurviger Format von Google Drive Cloud nicht mehr in Cruiser App auf mein Samsung S10e importieren (Discussion #450)


@RolfGS<https://github.com/RolfGS> thanks, now we can see the screenshots.

  *   The menu in image 2 appears when we select multiple routes (with tap) to do actions on all of them.
The selected routes then have a checkbox on their icon at the left.
This menu shows the number of the selected routes "Routen (1)".
  *   To display the menu in image 3 for a single route, press the menu button with the 3 dots at the right.
Or you can swipe the route to the right to load it.

Please let me know if you need more help.

**Reply by devemux86** [reply]
We discuss more improvements to the favorites screen, such as showing the menu and with long press anywhere on the items:
- https://github.com/devemux86/cruiser/discussions/555
